java/code/src/com/suse/manager/webui/controllers/contentmanagement/handlers/LivePatchingApiController.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2021 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.webui.controllers.contentmanagement.handlers;
+
+import com.redhat.rhn.domain.product.SUSEProduct;
+import com.redhat.rhn.domain.product.SUSEProductFactory;
+import com.redhat.rhn.domain.rhnpackage.PackageEvr;
+import com.redhat.rhn.domain.user.User;
+import com.suse.manager.model.products.Product;
+import com.suse.manager.webui.utils.gson.ResultJson;
+import org.apache.http.HttpStatus;
+import spark.Request;
+import spark.Response;
+import spark.Spark;
+
+import java.util.Comparator;
+import java.util.stream.Stream;
+
+import static com.suse.manager.webui.utils.SparkApplicationHelper.json;
+import static com.suse.manager.webui.utils.SparkApplicationHelper.withUser;
+import static java.util.stream.Collectors.toList;
+import static spark.Spark.get;
+
+/**
+ * Spark controller ContentManagement Live Patching API
+ */
+public class LivePatchingApiController {
+
+    private LivePatchingApiController() { }
+
+    /**
+     * JSON object for kernel versions
+     */
+    public static class KernelVersion {
+        private Long id;
+        private String version;
+
+        /**
+         * Initialize an object from an EVR
+         * @param evr the kernel EVR
+         */
+        public KernelVersion(PackageEvr evr) {
+            this.id = evr.getId();
+            this.version = evr.toUniversalEvrString();
+        }
+    }
+
+    /** Init routes for ContentManagement Live Patching Api.*/
+    public static void initRoutes() {
+        get("/manager/api/contentmanagement/livepatching/products",
+                withUser(LivePatchingApiController::getSuseProducts));
+        get("/manager/api/contentmanagement/livepatching/kernels/:productId",
+                withUser(LivePatchingApiController::getKernelVersions));
+    }
+
+    /**
+     * Get all SUSE root products that support live patching
+     *
+     * @param req the http request
+     * @param res the http response
+     * @param user the current user
+     * @return the JSON data
+     */
+    public static String getSuseProducts(Request req, Response res, User user) {
+            Stream<Product> suseProducts = SUSEProductFactory.getLivePatchSupportedProducts()
+                    .map(p -> new Product(p.getId(), null, p.getFriendlyName(), null, false, null, null, null));
```

Comment:
I wonder if it could be easier/more readable/more maintainable to create a new method to create a new instance in the `Product` class getting only parameters we have. But it can be seen as an OT comment to think about it later.